presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkConfig.java

```diff
@@ -42,6 +42,8 @@
     private double memoryRevokingThreshold;
     private double memoryRevokingTarget;
     private boolean retryOnOutOfMemoryBroadcastJoinEnabled;
+    private boolean retryOnOutOfMemoryWithIncreasedMemorySettingsEnabled;
+    private String sparkRetryIncreasedMemorySettingsJson = "{\"sparkSettings\":{},\"prestoSettings\":{}}";
```

Comment:
Will we have dynamic configs during retry? Like the oom retry, can we have the fixed override in `getRetryExecutionSettings` to avoid string encoding here?

Reply:
Yes. We can have dynamic configs that can be applied on retry based on what was the original configs that were used.   My initial design was to use static set of configs but then I realized that extending that would always require a code change and release. Hence, decided to use the json encoding here. This will enable us to have different set of configs and values for different kinds of containers deployment and it will be easily extensible.